File: app/src/main/java/com/seafile/seadroid2/framework/model/dirents/CachedDirentModel.java
package com.seafile.seadroid2.framework.model.dirents;

import androidx.annotation.Nullable;
import androidx.room.Embedded;

import com.seafile.seadroid2.framework.db.entities.DirentModel;

public class CachedDirentModel {
    @Embedded
    public DirentModel dirent;

    @Nullable
    public String local_file_id;
}

File: app/src/main/java/com/seafile/seadroid2/framework/model/dirents/CachedDirentModel.kt
package com.seafile.seadroid2.framework.model.dirents

import androidx.room.Embedded
import com.seafile.seadroid2.framework.db.entities.DirentModel

class CachedDirentModel {
    @Embedded
    @JvmField
    var dirent: DirentModel? = null

    @JvmField
    var local_file_id: String? = null
}
